packages/smooth_app/lib/pages/scan/scanner_state_manager.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:flutter/material.dart';
+import 'package:visibility_detector/visibility_detector.dart';
+
+/// This Widgets tracks if the scanner is currently visible and if the app
+/// is currently open/idle/closed and controls the camera depending
+class LifeCycleManager extends StatefulWidget {
+  const LifeCycleManager({
+    required this.onResume,
+    required this.onStop,
+    required this.child,
+    Key? key,
+  }) : super(key: key);
+
+  final Function() onResume;
+  final Function() onStop;
+  final Widget child;
+
+  @override
+  LifeCycleManagerState createState() => LifeCycleManagerState();
+}
+
+class LifeCycleManagerState extends State<LifeCycleManager>
+    with WidgetsBindingObserver {
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance!.addObserver(this);
+  }
+
+  @override
+  void dispose() {
+    WidgetsBinding.instance!.removeObserver(this);
+    super.dispose();
+  }
+
+  // Lifecycle changes are not handled by either of the used plugin. This means
+  // we are responsible to control camera resources when the lifecycle state is
+  // updated. Failure to do so might lead to unexpected behavior
+  // didChangeAppLifecycleState is called when the system puts the app in the
+  // background or returns the app to the foreground.
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    if (state == AppLifecycleState.inactive) {
+      widget.onStop.call();
+    } else if (state == AppLifecycleState.resumed) {
+      widget.onResume.call();
```

Comment:
is it possible for you to explain in comments when `AppLifeCycleState` and when `onVisibilityChanged` is called, both call `onStop` or `onResume` and right now as a reader I don't understand changes on the camera on device trigger what...

Reply:
For short, the visibility is if the scanner is currently displayed. So if it is the currently selected tab and it is also triggered if another page like search or the product page is above it.

The life cycle status is from the whole app. So this is triggered when the app is minimized and put back into focus again (useful for #982).

The two functions are just stopping and restarting the camera.